src/plugins/intel_gna/src/backend/gna_limitations.hpp

```diff
@@ -58,9 +59,41 @@ constexpr uint32_t bytesPerCropElement = 2;
 
 constexpr uint32_t kMemoryPageSize = 4096;
 
+class RuntimeTarget {
+private:
+    RuntimeTarget() {}
+
+    // static std::shared_ptr<Service> m_instance;
+    target::DeviceVersion m_compile_target = target::DeviceVersion::NotSet;
+    size_t m_mem_alignment = 0;
+
+public:
+    static RuntimeTarget& RuntimeTarget::GetInstance() {
```

Comment:
I see that there are methods in utils, layers and limitiations wich uses this new class in this PR.

This singleton require to be initialized before any method will use it,
I am afraid that it may lead to potential issues in unit tests and in code it self.

Default constructor of the GNAPlugin (used by properties) does not initialize device.

Maybe there is possibility to create instance of this class after device is iniitlized and pass reference or pointer to its instance later where it is needed.

Reply:
do_not_review label - It is not supposed to be reviewed